source/Engine.cpp

```diff
@@ -2052,17 +2052,20 @@ void Engine::DoCollisions(Projectile &projectile)
 			for(Body *body : shipCollisions.Circle(hitPos, blastRadius))
 			{
 				Ship *ship = reinterpret_cast<Ship *>(body);
-				if(isSafe && projectile.Target() != ship && !gov->IsEnemy(ship->GetGovernment()))
+				bool targeted = (projectile.Target() == ship);
+				if(isSafe && !targeted && !gov->IsEnemy(ship->GetGovernment()))
 					continue;
 
-				int eventType = ship->TakeDamage(visuals, damage.CalculateDamage(*ship, ship == hit.get()), projectile.GetGovernment());
+				int eventType = ship->TakeDamage(visuals, damage.CalculateDamage(*ship, ship == hit.get()),
+					gov, targeted);
```

Comment:
```suggestion
				int eventType = ship->TakeDamage(visuals, damage.CalculateDamage(*ship, ship == hit.get()), gov, targeted);
```
It would be nice to do something like `!isBlast && targeted ? gov : nullptr` but there may be cases that pattern doesn't support.

It's possible we could make it work, though, since all call sites know whether the damage is due to blast damage or not.

Reply:
DamageDealt only contains isBlast so that TakeDamage knows if the ship should become provoked, so if we did make such a change then I don't see why DamageDealt would need to contain the isBlast bool.